src/MediatR/AsyncRequestHandler.cs

```diff
@@ -11,7 +11,7 @@ public abstract class AsyncRequestHandler<TMessage> : IAsyncRequestHandler<TMess
     {
         public async Task<Unit> Handle(TMessage message)
         {
-            await HandleCore(message);
+            await HandleCore(message).ConfigureAwait(false);
```

Comment:
In your case this is not really achievable without introducing unnecessary complexity. I'd rather go and introduce another interface on the library level like `IAsyncRequestHandler<TMessage>` and let the mediator handle the smarts